src/EnablerExtension.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace SilverStripe\LoginForms;
+
+use SilverStripe\View\SSViewer;
+use SilverStripe\Core\Extension;
+use SilverStripe\Security\Security;
+use SilverStripe\Core\Config\Config;
+
+/**
+ * Applies to the {@see Security} controller in order to detect requests for actions related to
+ * the log in process or other such credential management (such as the forgot password flow).
+ * This is in order to replace the set {@see SSViewer} theme list with a controlled set in order
+ * to always show a consistent interface that relates more to the CMS than the website it is
+ * loading on.
+ * Particular actions can be set to be ignored by inlcuding them in the `excluded_actions` list
+ * defined in yml _config for this class. By default all allowed actions on the Security controller
+ * excepting `index` and `ping` will have the new theme set applied.
+ */
+class EnablerExtension extends Extension
+{
+    public function beforeCallActionHandler()
+    {
+        $config = Config::inst();
+        /** @var Security */
+        $owner = $this->owner;
+        $action = $owner->getAction();
+        $allowedActions = $config->get(Security::class, 'allowed_actions');
+        $excludedActions = $config->get(self::class, 'excluded_actions');
+        $themeActions = array_diff($allowedActions, $excludedActions);
+        if (in_array($action, $themeActions))
+        {
```

Comment:
PHPCS linting error. Can you ensure you add unit tests and phpcs checks to the repo?

Reply:
Oh whoops, haha. I did this on purpose to test the linter integration with my IDE (it's broken 😆).